examples/fizzbuzz/main.c

```diff
@@ -0,0 +1,189 @@
+                                    0, NULL);
+        strcpy(fb->result, "FizzBuzz");
+    } else if (i % 3 == 0) {
+        strcpy(fb->result, "Fizz");
+    } else if (i % 5 == 0) {
+        strcpy(fb->result, "Buzz");
+    } else {
+        sprintf(fb->result, "%d", i);
+    }
+
+    qmlbind_value params[1] = { qmlbind_value_new_string(strlen(fb->result), fb->result) };
+    qmlbind_signal_emitter_emit(fb->signalEmitter, "resultChanged", 1, params);
+}
+
+
+qmlbind_client_object *
+fizzbuzz_new_object_cb(qmlbind_client_class *classRef,
+                       qmlbind_signal_emitter *signalEmitter)
+{
+    fizzbuzz_t *fb = malloc(sizeof(fizzbuzz_t));
+    fb->input = 0;
+    fb->result = malloc(sizeof(char) * ENOUGH);
+    strcpy(fb->result, "");
+    fb->signalEmitter = signalEmitter;
+
+    return (qmlbind_client_object *) fb;
+}
+
+
+void fizzbuzz_delete_object_cb(qmlbind_client_object *obj)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+    free(fb->result);
+    qmlbind_signal_emitter_release(fb->signalEmitter);
+    free(fb);
+}
+
+
+qmlbind_value *fizzbuzz_call_method_cb(qmlbind_engine *engine,
+                                       qmlbind_client_object *obj,
+                                       const char *name,
+                                       int argc,
+                                       const qmlbind_value *const *argv)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "onInputChanged") == 0 && argc == 1) {
+        on_changed(fb, argv[0]);
+        return qmlbind_value_new_undefined();
+    } else if (strcmp(name, "quit") == 0) {
+        exit(0); // FIXME: should call app.quit()
+    } else {
+        return qmlbind_value_new_undefined();
+    }
+}
+
+
+qmlbind_value *fizzbuzz_get_property_cb(qmlbind_engine *engine,
+                                        qmlbind_client_object *obj,
+                                        const char *name)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "input")) {
+        return qmlbind_value_new_number(fb->input);
+    } else if (strcmp(name, "result")) {
+        return qmlbind_value_new_string(strlen(fb->result), fb->result);
+    } else {
+        return qmlbind_value_new_null();
+    }
+}
+
+
+void fizzbuzz_set_property_cb(qmlbind_engine *engine,
+                              qmlbind_client_object *obj,
+                              const char *name,
+                              const qmlbind_value *value)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "input")) {
+        fb->input = (int) qmlbind_value_get_number(value);
+
+        qmlbind_value params[1] = { qmlbind_value_new_number(fb->input) };
+        qmlbind_signal_emitter_emit(fb->signalEmitter, "inputChanged", 1, params);
+    } else if (strcmp(name, "result")) {
+        qmlbind_string *qstr = qmlbind_value_get_string(value);
+        const char *str = qmlbind_string_get_chars(qstr);
+        strcpy(fb->result, str);
```

Comment:
`qstr` is owned by the client code, you need to release it after copying it. Which was also missing, seems like I overlooked it because all other `qmlbind_value_get_*` return primitive types.